include/toml++/impl/path.h

```diff
@@ -0,0 +1,438 @@
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path leaf(size_t n = 1) const;
+
+		/// \brief	Appends elements to the end of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(const toml::path&);
+
+		/// \brief	Appends elements to the end of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(toml::path&&);
+
+		/// \brief	Appends elements to the end of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(std::string_view);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(const toml::path&);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(toml::path&&);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(std::string_view);
+
+		/// \brief	Replaces the contents of the path object by a new path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& assign(std::string_view);
+
+		/// \brief	Replaces the contents of the path object by a new path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& assign(const path&);
+
+		/// \brief Returns a string representing the path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		TOML_NODISCARD
+		std::string string() const;
+
+#if TOML_ENABLE_WINDOWS_COMPAT
+
+		/// \brief Construct from wstring
+		TOML_NODISCARD_CTOR
+		TOML_EXPORTED_MEMBER_FUNCTION
+		explicit path(std::wstring_view);
+
+		/// \brief	Append operator
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& operator/=(std::wstring_view);
+
+		/// \brief	Append operator.
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& operator+=(std::wstring_view);
+
+		/// \brief Implicitly cast to a std::wstring
+		inline operator std::wstring() const
+		{
+			return wstring();
+		}
+
+		/// \brief Evaluate whether two paths are the same
+		TOML_EXPORTED_MEMBER_FUNCTION
+		bool operator==(std::wstring_view compare) const noexcept;
+
+		/// \brief Evaluate whether two paths are the same
+		TOML_EXPORTED_MEMBER_FUNCTION
+		bool operator!=(std::wstring_view compare) const noexcept;
+
+		/// \brief	Appends elements to the end of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(std::wstring_view);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(std::wstring_view);
+
+		/// \brief	Replaces the contents of the path object by a new path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& assign(std::wstring_view);
+
+		/// \brief Returns a wstring representing the path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		std::wstring wstring() const;
+
+#endif // TOML_ENABLE_WINDOWS_COMPAT
+
+		// === Hidden friend operators ========================================
+		friend std::ostream& operator<<(std::ostream& os, const toml::path& rhs)
+		{
+			os << rhs.string();
+			return os;
+		}
+
+		friend std::istream& operator>>(std::istream& is, toml::path& rhs)
+		{
+			std::string s;
+			is >> s;
+			rhs.assign(s);
+
+			return is;
+		}
```

Comment:
The implementations for these two functions should be moved to the `.inl` file since they'll require `#include <ostream>` and `#include <istream>`, something I only do in the `TOML_IMPLEMENTATION` parts of the library so that people who disable `TOML_HEADER_ONLY` can pay the cost of these includes only once. (I already `#include <iosfwd>` in `forward_declarations.h`)

Same goes for any functions that use stringstreams, if there are any.

Also, if you haven't seen it already, let me introduce you to a great "how expensive is this header" tool: https://artificial-mind.net/projects/compile-health/ - type "stream" into the search bar and be shocked :D

Reply:
Terrifying to include filestream or complex or regex anywhere!